tests/gtest/avifgainmaptest.cc

```diff
@@ -933,6 +933,51 @@ class ToneMapTest
           /*reference=*/std::string, /*min_psnr=*/float, /*max_psnr=*/float>> {
 };
 
+void ToneMapImageAndCompareToReference(
+    const avifImage* base_image, const avifGainMap& gain_map,
+    float hdr_headroom, int out_depth,
+    avifTransferCharacteristics out_transfer_characteristics,
+    avifRGBFormat out_rgb_format, const avifImage* reference_image,
+    float min_psnr, float max_psnr, float* psnr_out = nullptr) {
+  SCOPED_TRACE("hdr_headroom: " + std::to_string(hdr_headroom));
+
+  testutil::AvifRgbImage tone_mapped_rgb(base_image, out_depth, out_rgb_format);
+  testutil::AvifImagePtr tone_mapped(
+      avifImageCreate(tone_mapped_rgb.width, tone_mapped_rgb.height,
+                      tone_mapped_rgb.depth, AVIF_PIXEL_FORMAT_YUV444),
+      avifImageDestroy);
+  tone_mapped->transferCharacteristics = out_transfer_characteristics;
+  tone_mapped->colorPrimaries = base_image->colorPrimaries;
+  tone_mapped->matrixCoefficients = base_image->matrixCoefficients;
+
+  avifDiagnostics diag;
+  avifResult result = avifImageApplyGainMap(
+      base_image, &gain_map, hdr_headroom, tone_mapped->transferCharacteristics,
+      &tone_mapped_rgb, &tone_mapped->clli, &diag);
+  ASSERT_EQ(result, AVIF_RESULT_OK)
+      << avifResultToString(result) << " " << diag.error;
+  ASSERT_EQ(avifImageRGBToYUV(tone_mapped.get(), &tone_mapped_rgb),
+            AVIF_RESULT_OK);
+  if (reference_image != nullptr) {
+    EXPECT_EQ(out_depth, (int)reference_image->depth);
+    const double psnr = testutil::GetPsnr(*reference_image, *tone_mapped);
+    printf("PSNR: %f\n", psnr);
```

Comment:
Leftover?

Reply:
It was intentional, yes it's debugging but I think it's useful to have it there and it doesn't hurt...

Reply:
What about `SCOPED_TRACE` then? Does it print only on failure? Also consider `std::cout` instead of `printf`.

Reply:
Yes SCOPED_TRACE only prints on failure. Changed to std::cout.